qwandaq/src/main/java/life/genny/qwandaq/utils/product/SearchBuilder.java

```diff
@@ -0,0 +1,120 @@
+ * 
+ * @author Bryn Meachem
+ */
+@ApplicationScoped
+public class SearchBuilder {
+
+    @Inject
+    Logger log;
+
+    @Inject
+    CacheManager cm;
+
+    @ConfigProperty(name = "genny.product")
+    String CONFIG_PRODUCT_CODE;
+    
+    public SearchBuilder() { /* no-args constructor */ }
+
+    /**
+     * Create a new Dropdown SearchEntity
+     * @param dropdownCode - {@link BaseEntity#getCode() baseentity code} for the dropdown SearchEntity
+     * @param filters - filters to apply to get the base entities to return from teh dropdown SearchEntity
+     * @return a dropdown SearchEntity with the filters supplied and the name set to the supplied dropdownCode
+     */
+    public SearchEntity dropdown(String dropdownCode, Filter... filters) {
+        return dropdown(dropdownCode, dropdownCode, filters);
+    }
+
+    /**
+     * Create a new Dropdown SearchEntity
+     * @param dropdownCode - {@link BaseEntity#getCode() baseentity code} for the dropdown SearchEntity
+     * @param selectionParentCode - the parent code that all selections have as their {@link Attribute#LNK_PARENT} (for single filter dropdowns)
+     * @return - a dropdown SearchEntity for the given parent code with the name set to the supplied dropdownCode
+     */
+    public SearchEntity dropdown(String dropdownCode, String selectionParentCode) {
+        return dropdown(dropdownCode, dropdownCode, selectionParentCode);
+    }
+
+    /**
+     * Create a new Dropdown SearchEntity
+     * @param dropdownCode - {@link BaseEntity#getCode() baseentity code} for the dropdown SearchEntity
+     * @param dropdownName - {@link BaseEntity#getName() name} of the dropdown
+     * @param selectionParentCode - the parent code that all selections have as their {@link Attribute#LNK_PARENT} (for single filter dropdowns)
+     * @return - a dropdown SearchEntity for the given parent code
+     */
+    public SearchEntity dropdown(String dropdownCode, String dropdownName, String selectionParentCode) {
+        return dropdown(dropdownCode, dropdownName, new Filter(Attribute.LNK_PARENT, Operator.CONTAINS, selectionParentCode));
+    }
+
+    /**
+     * Create a new Dropdown SearchEntity
+     * @param dropdownCode - {@link BaseEntity#getCode() baseentity code} for the dropdown SearchEntity
+     * @param dropdownName - {@link BaseEntity#getName() name} of the dropdown
+     * @param filters - filters to apply to get the base entities to return from teh dropdown SearchEntity
+     * @return a dropdown SearchEntity with the filters supplied
+     */
+    public SearchEntity dropdown(String dropdownCode, String dropdownName, Filter... filters) {
+        if(!dropdownCode.startsWith(Prefix.SBE_))
+            dropdownCode = Prefix.SBE_.concat(dropdownCode);
+        
+        SearchEntity dropdown = new SearchEntity(dropdownCode, dropdownName);
+        for(Filter filter : filters) {
+            if(filter.getCode().startsWith("PRI_IS_")) {
```

Comment:
We should discuss what we want to do with PRI_IS. There are dropdowns currently being created that still use it and if we don't decide on it now I am speculating we will end up stuck with it in the future